From 142fc9f0459ab0fb066339af45cf2fadb9b2c7b9 Mon Sep 17 00:00:00 2001
From: Fabian Kosmale <fabian.kosmale@qt.io>
Date: Mon, 6 Oct 2025 14:05:51 +0200
Subject: [PATCH 164/239] CRA: Review local storage

Mark it as significant. The code that accesses local storage needs to be
trusted (like any QML code), and the actual work is carried out by the
QSql classes.

Fixes: QTBUG-136200
Pick-to: 6.8
Change-Id: I21c129b1ad49011330062efdc3307753ce7461a4
Reviewed-by: Sami Shalayel <sami.shalayel@qt.io>
(cherry picked from commit 4e8bd3111990ee1f4a6bbfe1d1a767520e4c3e11)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/qmllocalstorage/qqmllocalstorage.cpp       | 1 +
 src/qmllocalstorage/qqmllocalstorage_p.h       | 1 +
 src/qmllocalstorage/qqmllocalstorageglobal_p.h | 1 +
 3 files changed, 3 insertions(+)

diff --git a/src/qmllocalstorage/qqmllocalstorage.cpp b/src/qmllocalstorage/qqmllocalstorage.cpp
index bed86c141e..f04017daf6 100644
--- a/src/qmllocalstorage/qqmllocalstorage.cpp
+++ b/src/qmllocalstorage/qqmllocalstorage.cpp
@@ -1,5 +1,6 @@
 // Copyright (C) 2019 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #include "qqmllocalstorage_p.h"
 
diff --git a/src/qmllocalstorage/qqmllocalstorage_p.h b/src/qmllocalstorage/qqmllocalstorage_p.h
index 510ee78dbe..4e987529fb 100644
--- a/src/qmllocalstorage/qqmllocalstorage_p.h
+++ b/src/qmllocalstorage/qqmllocalstorage_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2019 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:significant
 
 #ifndef QQMLLOCALSTORAGE_P_H
 #define QQMLLOCALSTORAGE_P_H
diff --git a/src/qmllocalstorage/qqmllocalstorageglobal_p.h b/src/qmllocalstorage/qqmllocalstorageglobal_p.h
index 9f36a1f367..bd8c27d0eb 100644
--- a/src/qmllocalstorage/qqmllocalstorageglobal_p.h
+++ b/src/qmllocalstorage/qqmllocalstorageglobal_p.h
@@ -1,5 +1,6 @@
 // Copyright (C) 2021 The Qt Company Ltd.
 // SPDX-License-Identifier: LicenseRef-Qt-Commercial OR LGPL-3.0-only OR GPL-2.0-only OR GPL-3.0-only
+// Qt-Security score:insignificant
 
 #ifndef QQMLLOCALSTORAGEGLOBAL_P_H
 #define QQMLLOCALSTORAGEGLOBAL_P_H
-- 
2.51.2

